go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/cluster-capi-operator
 
-go 1.21
+go 1.22
```

Comment:
We have a Go builder for 4.17, so I don't think it's worth jumping through hoops to try to avoid this bump. CAPO v0.10 uses go 1.22.